### ISSUE TITLE
Suggestions #1

### ISSUE THREAD

**Issue by JjTheHusky** [SteffenL/dise#9]
Just a suggestions thread, these are personal and up to the developer if they want to do anything with them.

- My first suggestion is to add a button to launch the game from the editor. This is just a 'nice touch' kinda thing, or for people to use if they are lazy to not open steam.

- Secondly, a way to access developer stuff? I heard that developers in the game have sometimes left tools such as noclip and dev menu's in some updates, could it be possible to get it to occur with the editor?

- Ability to adjust game settings from the editor. Can be graphical or even hidden options if there are any.

- More descriptive Mod Blueprints. Mainly around developer ones, describing what they can be used on and what it does in-game.

**Comment by SteffenL** [comment]
Thanks for the suggestions!

DISE is primarily a tool for modifying saved games. It was never meant for other purposes but that doesn't mean that it can't ever do anything else if people really want it.

> launch the game from the editor

I can see that being useful in some way as long as we can fit that in the user interface. It should take no more than one click (or keyboard shortcut) to launch the game. I think we can get away with showing e.g. a single menu entry like "Launch game" when a saved game is open, and that would launch the game that corresponds to that file.

> access developer stuff

The developer menu can probably be unlocked but it would require modding the game itself and not the game-save file. It's a different territory and it would need some planning to implement well.

> adjust game settings from the editor

I am not quite sure how useful something like that would be. Is it different from duplicating what the game can already do?

> More descriptive Mod Blueprints

If this is available in the game then it would be possible to extract this information. If not then we will have to get creative. The "Mods/Collectibles" tab in DISE shows some stats but not full descriptions.

**Comment by Wahkah** [comment]
The in game settings lacks control over shadow map sizing, ambient occlusions, bloom, and texture sizing fit for lower end conputers. An example would be setting it to 32x32 for shadows or 64x64 and that would tremendouly increase game frames. I believe the default min is 256x256 and max is 1024x1024. This may even give preference for high end computers who can handle 2048x2048 and 4096x4096 resolutions.

**Comment by Wahkah** [comment]
I’ll go ahead and extract those blueprint files data bins for Steffen.

**Comment by JjTheHusky** [comment]
> > access developer stuff
> 
> The developer menu can probably be unlocked but it would require modding the game itself and not the game-save file. It's a different territory and it would need some planning to implement well.

Absolutely, I can't find anything regarding accessing the dev stuff. This could possibly be looked at a later time.

> > adjust game settings from the editor
> 
> I am not quite sure how useful something like that would be. Is it different from duplicating what the game can already do?

Basically having functionality to edit the ingame settings from the editor. This could easily be done by modifying the options config file, wherever it is. Wahkah explains how this could be useful to change hidden values that aren't show ingame.

> > More descriptive Mod Blueprints
> 
> If this is available in the game then it would be possible to extract this information. If not then we will have to get creative. The "Mods/Collectibles" tab in DISE shows some stats but not full descriptions.

This seems to already be available in-game, as shown here.
![image](https://user-images.githubusercontent.com/25309518/58438454-9a139300-8094-11e9-97ed-aa1f7e1aa79b.png)
This should work just fine if it was possible to extract it from the game files. I would look around but I don't know how to look around in the data packs.

**Comment by Wahkah** [comment]
Already extracted and sent to Steffen!
(Blueprints aka craftplans)

**Comment by Wahkah** [comment]
Nvm he got it! 😂

**Comment by JjTheHusky** [comment]
I looked around at dev menu stuff, and it's possible. For the OG game, it's as simple as replacing a DLL in the game folder, found one for DE but it looks like it involves .pak modification, looking into it.

https://youtu.be/6tgdoMwcDSA?t=140

**Comment by Wahkah** [comment]
The offsets for both DE and OG i have to convert the dlls to UseDeveloperKeys(). 
For the dev menus, i made most of the ones back in the day. My youtube was Die12107 but removed the videos after archiving the videos since i was going to post it all on Nexus.

**Comment by JjTheHusky** [comment]
Small suggestion, adding an icon to appear in the apps list for windows.
![ApplicationFrameHost_jPLp7pvf7T](https://user-images.githubusercontent.com/25309518/58516313-23969400-816d-11e9-9bc5-675d2057b229.png)

**Comment by SteffenL** [comment]
Added icon.

![image](https://user-images.githubusercontent.com/1543854/58566686-5904b580-8231-11e9-9698-7387f5cff7c1.png)

**Comment by JjTheHusky** [comment]
I know it's quiet here, but just a quick suggestion;
A debug/console log for DISE for reporting bugs/issues.
